From c3efe0c576c59a648d1965dc1604960529adc7b5 Mon Sep 17 00:00:00 2001
From: Simon Dubray <simonx.dubray@intel.com>
Date: Tue, 26 May 2015 10:52:40 +0200
Subject: [PATCH] alarmtimer: adjust the duration of __pm_wakeup_event

When an alarm expires in the two next seconds, alarmtimer aborts the suspend
and uses a wakeup event to prevent suspend during two seconds.
Because of this, we are seing big wakeups (about 3 seconds instead of 1).

Change-Id: I2f0890ffcfe3ad6f4ce4544a4d78f0796ed844a7
Signed-off-by: Simon Dubray <simonx.dubray@intel.com>
Signed-off-by: Charly Dumur <charly.dumur@intel.com>
---
 kernel/time/alarmtimer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/time/alarmtimer.c b/kernel/time/alarmtimer.c
index cd45a0727a16..4eeb130dbe0e 100644
--- a/kernel/time/alarmtimer.c
+++ b/kernel/time/alarmtimer.c
@@ -255,7 +255,7 @@ static int alarmtimer_suspend(struct device *dev)
 		return 0;
 
 	if (ktime_to_ns(min) < 2 * NSEC_PER_SEC) {
-		__pm_wakeup_event(ws, 2 * MSEC_PER_SEC);
+		__pm_wakeup_event(ws, ktime_to_ms(min) + 1);
 		return -EBUSY;
 	}
 
-- 
1.9.1

